### ISSUE TITLE
Urgently migrate sfdx source:convert and mdapi:deploy used in local compile to latest sf cli format

### ISSUE THREAD

**Issue by kgo20240829engie** [vlocityinc/vlocity_build#687]
Salesforce itself, so your colleagues will remove some commands used by vlocity Build Tool in the localCompile process. see https://github.com/forcedotcom/cli/issues/2974

There is already an alpha work done by vlocity R&D team to migrate to using pure SF commands but not reaching beta or release yet.

Please release things early enough to not have pipelines using localCompile to simply crash because of this communication and alignment mismatch between teams.

**Comment by calvinle1** [comment]
Hi, is there an update to this? Was initially notified that the team is working on this but have not received updates since.

**Comment by AbhinavKumar-sf** [comment]
@kgo20240829engie @calvinle1  @thvd  in which code flow you see the usage of this command, we are including **alm** as node dependency itself

**Comment by kgo20240829engie** [comment]
Hi @AbhinavKumar-sf , I don't understand your answer
Firstly, alm is deprecated and now completely unmaintained
Secondly, you offer the possibility to use our own CLI (see your own readme https://github.com/vlocityinc/vlocity_build?tab=readme-ov-file#how-to-use-your-own-sfdx-cli-to-deploy-omniscripts--flexcards-compiled-locally)

Whatever the reason, you should make sure to analyse potential problems and not let old deprecated packages rot, especially since they are colleagues building and maintaining them.

**Comment by kgo20240829engie** [comment]
I could add that you already refuse to improve puppeteer order of LWC activation (see #521).
Then the only option clients can use is local compile which is either using a deprecated package that might fail without warning or an up to date CLI that refuses to work because you didn't update the commands used.

**Comment by AbhinavKumar-sf** [comment]
@kgo20240829engie  we have introduced new flag (**useSfCli** - true)  for  executing sf commands in case of local compilation. These changes are already available in [beta](https://www.npmjs.com/package/vlocity/v/1.17.8-beta.1) . If you want you can give it a try.

**Comment by thvd** [comment]
Well, here we go, SF CLI 2.65.8 is published, and current stable version of VBT is broken...

**Comment by thvd** [comment]
Still get this error with that beta version [1.17.8-beta.1](https://www.npmjs.com/package/vlocity/v/1.17.8-beta.1)

```
Command to be executed is  >> sfdx force:mdapi:deploy --deploydir "vlocity-temp/mdapi" --targetusername "abc@abbbccc.com.acc.b2b" -w -1 --verbose --json  
  Error: Error parsing stdout
    at _run (/home/vsts/work/1/s/node_modules/vlocity/lib/sfdx.js:13:9)
    at _mdapiDeploy (/home/vsts/work/1/s/node_modules/vlocity/lib/sfdx.js:35:22)
    at run (/home/vsts/work/1/s/node_modules/vlocity/lib/sfdx.js:43:10)
    at DataPacksJob.deployJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:2809:49)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DataPacksJob.doRunJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:339:9)
    at async DataPacksJob.deployJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:2888:13)
    at async DataPacksJob.doRunJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:339:9)
    at async DataPacksJob.runJobWithInfo (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:293:9)
    at async DataPacksJob.runJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:95:26)  
```

**Comment by KevinGossentCap** [comment]
@thvd , from what I read in the commits, try setting the usesfcli parameter to true instead of sesfdxcli.
Hoping that it helps with the beta.
But yes they are making life hard making the compatibility development so late, not documented and not sure to have been explicitly tested

**Comment by thvd** [comment]
@KevinGossentCap You are right, my mistake and I forgot about that, was too much in a hurry. 
I am now testing with that new flag and using the new beta version

**Comment by KevinGossentCap** [comment]
@thvd your other option is to lock the SF cli version in your pipeline to last week's version and use official CBT version

**Comment by thvd** [comment]
I have run the beta version now twice in a pipeline, both times i got timeouts, seems unreliable: 

```
    "message": "The command has timed out, although the deployment is still running. Use \"sf project deploy resume\" to resume watching the deployment.",

```

That's with this stack trace:

```
  Command to be executed is  >>  sf project convert source --root-dir "vlocity-temp/tempDeployLWC/salesforce_sfdx" --output-dir "vlocity-temp/mdapi" --json  
  source:convert ended successfully  
  Running mdapi:deploy  
  Command to be executed is  >> sf project deploy start --metadata-dir "vlocity-temp/mdapi" --target-org "abc@abbcc.com.acc.b2b" -w 1 --verbose --json  
  Error: Command responded with error: sf project deploy start --metadata-dir "vlocity-temp/mdapi" --target-org "abc@abbcc.com.acc.b2b" -w 1 --verbose --json
    at _run (/home/vsts/work/1/s/node_modules/vlocity/lib/sfdx.js:17:17)
    at _mdapiDeploy (/home/vsts/work/1/s/node_modules/vlocity/lib/sfdx.js:36:2)
    at run (/home/vsts/work/1/s/node_modules/vlocity/lib/sfdx.js:43:10)
    at DataPacksJob.deployJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:2809:49)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DataPacksJob.doRunJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:339:9)
    at async DataPacksJob.deployJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:2888:13)
    at async DataPacksJob.doRunJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:339:9)
    at async DataPacksJob.runJobWithInfo (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:293:9)
    at async DataPacksJob.runJob (/home/vsts/work/1/s/node_modules/vlocity/lib/datapacksjob.js:95:26)  

```

**Comment by AbhinavKumar-sf** [comment]
@thvd Thanks for fixing the timeout issue.  it's available in vlocity@1.17.8-beta.2

**Comment by thvd** [comment]
@AbhinavKumar-sf Thanks, i will test later today and report back

**Comment by thvd** [comment]
@AbhinavKumar-sf I have run the first test, and that one was successful. Will keep an eye open on it, and if anything goes wrong will report here.

**Comment by dschibster** [comment]
@thvd @AbhinavKumar-sf for me, the beta version has not made a difference. The logs state the following:

```
 Deploying generated LWC components  
  Using sfdc cli  
  Running source:convert  
  Command to be executed is  >> sfdx force:source:convert --rootdir "vlocity-temp/tempDeployLWC/salesforce_sfdx" --outputdir "vlocity-temp/mdapi" --json  
  source:convert ended successfully  
  Running mdapi:deploy  
  Command to be executed is  >> sfdx force:mdapi:deploy --deploydir "vlocity-temp/mdapi" --targetusername "test-ctrhfwjxsh7h@example.com" -w -1 --verbose --json  
  Skipping Deactivation of OmniScript as keepOSActiveWhenLWCDeploymentFails = true. As per this param, Deployed OmniScripts will remain active irrespective of LWC is deployed or not.  
  Back to Ready  
  Deploying Vlocity  
  Adding to Deploy >> DataRaptor/CPQGetOrderMembers - CPQGetOrderMembers    
  Deploy >> 1 Items  
  Adding to Deploy >> DataRaptor/CPQLockGroupandMembers - CPQLockGroupandMembers    
  Deploy >> 1 Items  
  Deploying >> DataRaptor/CPQGetOrderMembers  
  Deploying >> DataRaptor/CPQLockGroupandMembers  
  Deploy Success >> DataRaptor/CPQGetOrderMembers CPQGetOrderMembers   
  Deploy Success >> DataRaptor/CPQLockGroupandMembers CPQLockGroupandMembers   
  Salesforce Org >> test-ctrhfwjxsh7h@example.com  
  Version Info >> v1.17.8-beta.2 vlocity_cmt 900.539  
  Current Status >> Deploy  
  Remaining >> 0  
  Success >> 1411  
  Elapsed Time >> 6m 4s  
  Migrated 1411 items >> Deploying  
  Salesforce Org >> test-ctrhfwjxsh7h@example.com  
  Version Info >> v1.17.8-beta.2 vlocity_cmt 900.539  
  Current Status >> Deploy  
  Remaining >> 0  
  Success >> 1411  
  Elapsed Time >> 6m 4s  
  Migrated 1411 items >> Deploying  
  Deploying generated LWC components  
  Using sfdc cli  
  Running source:convert  
  Command to be executed is  >> sfdx force:source:convert --rootdir "vlocity-temp/tempDeployLWC/salesforce_sfdx" --outputdir "vlocity-temp/mdapi" --json  
  source:convert ended successfully  
  Running mdapi:deploy  
  Command to be executed is  >> sfdx force:mdapi:deploy --deploydir "vlocity-temp/mdapi" --targetusername "test-ctrhfwjxsh7h@example.com" -w -1 --verbose --json  
  Skipping Deactivation of OmniScript as keepOSActiveWhenLWCDeploymentFails = true. As per this param, Deployed OmniScripts will remain active irrespective of LWC is deployed or not.  
  Finalizing Deployment  
  Finalizing Deployment  
  Salesforce Org >> test-ctrhfwjxsh7h@example.com  
  Version Info >> v1.17.8-beta.2 vlocity_cmt 900.539  
  Current Status >> Deploy  
  Remaining >> 0  
  Success >> 1411  
  Elapsed Time >> 6m 19s  
  Migrated 1411 items >> Deploying  
  Job Complete >> Deploy 6m 19s  
```

I have made sure that I use `useSfCli` instead of `useSfdxCli` in my yaml file, but still no difference, no locally compiled components are deployed.

Not sure if I am doing something wrong, but there are clear indications that the local compilation process itself is still working.

**Comment by dschibster** [comment]
Second test was successful, I was still on a previous version of my yaml. beta2 is working. :)

**Comment by calvinle1** [comment]
@thvd You can check for the npm releases here: https://www.npmjs.com/package/vlocity/v/1.17.10?activeTab=versions

Looks like 1.17.10 has the fix

**Comment by AbhinavKumar-sf** [comment]
This is fixed in 1.17.10

**Comment by KumaranM2040** [comment]
Can the README.md file be updated to reflect that **useSfCli: true** is the only option available ?

![Image](https://github.com/user-attachments/assets/182e5b85-65cd-4c52-ac59-3d585028714e)
